apps/wallet-mobile/src/components/NftPreview/NftPreview.tsx

```diff
@@ -12,38 +13,46 @@ import {useNativeAssetImage} from '../../yoroi-wallets/hooks'
 type NftPreviewProps = {
   nft: Balance.TokenInfo
   showPlaceholder?: boolean
-  style?: StyleProp<ImageStyle>
+  style?: ImageStyle
   height: number
   width: number
-  resizeMode?: 'cover' | 'contain'
+  contentFit?: 'cover' | 'contain'
   blurRadius?: number
-  zoom?: 1 | 2 | 3
+  zoom?: number
+  cachePolicy?: 'none' | 'memory' | 'disk' | 'memory-disk'
 }
 
 export const NftPreview = ({
   nft,
   showPlaceholder,
-  style,
+  style = {},
   height,
   width,
-  resizeMode = 'cover',
+  contentFit = 'cover',
   blurRadius,
   zoom = 1,
+  cachePolicy,
 }: NftPreviewProps) => {
   const wallet = useSelectedWallet()
   const [policy, name] = nft.id.split('.')
-  const {uri, isLoading, isError} = useNativeAssetImage({
+  const {uri, headers, isLoading, isError} = useNativeAssetImage({
     networkId: wallet.networkId,
     policy,
     name,
     width: width * zoom,
     height: height * zoom,
     kind: 'metadata',
-    resizeMode,
+    contentFit,
     mediaType: getNftMainImageMediaType(nft),
   })
 
-  const shouldShowPlaceholder = !isString(uri) || showPlaceholder || isError
+  const [failed, setFailed] = useState(false)
+
+  useEffect(() => {
+    setFailed(false)
+  }, [uri])
```

Comment:
Could `onSuccess` or `onSettle` from `useNativeAssetImage` be useful?

Reply:
well callbacks are deprecated in new versions so i guess is the best option

Reply:
I think it's ok cuz react-query is dropping on the next version all callbacks.

Reply:
`useNativeAssetImage` doesn't have those. Would have to create them and pass them to useQuery. But since newer versions of useQuery don't support that, seems like adding something that will need to be removed whenever react-query gets updated

Reply:
Yeah that was my initial idea but i remembered this

https://tkdodo.eu/blog/breaking-react-querys-api-on-purpose